testing/trino-testing/src/main/java/io/trino/testing/PlanDeterminismChecker.java

```diff
@@ -49,13 +48,15 @@ public void checkPlanIsDeterministic(String sql)
 
     public void checkPlanIsDeterministic(Session session, String sql)
     {
-        IntStream.range(1, MINIMUM_SUBSEQUENT_SAME_PLANS)
-                .mapToObj(attempt -> getPlanText(session, sql))
-                .map(planEquivalenceFunction)
-                .reduce((previous, current) -> {
-                    assertEquals(previous, current);
-                    return current;
-                });
+        String previous = null;
+        for (int attempt = 1; attempt < MINIMUM_SUBSEQUENT_SAME_PLANS; attempt++) {
+            String planText = getPlanText(session, sql);
+            String current = planEquivalenceFunction.apply(planText);
+            if (previous != null) {
```

Comment:
Base it on `attempt` value, so you don't ignore it when `previous` is null on not the first invocation.

Or perhaps init the previous before the loop body (and rename):
```
String first = planEquivalenceFunction.apply(getPlanText(session, sql));
```

Reply:
I took the latter approach, which also fixes the apparent off-by-one in the number of iterations.